handbook/repository/aws-secrets.md

```diff
@@ -0,0 +1,123 @@
+# AWS Secrets
+
+## Prerequisites
+
+- You will need [AWS command line](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed.
+- You will also need access to the AWS account. Ask someone from DevOps to send you an invitation.
+
+## Getting started
+
+1. Configure AWS with SSO, using the following configuration:
+
+```bash
+aws configure sso
+
+SSO start URL [None]: https://island-is.awsapps.com/start
+SSO Region [None]: eu-west-1
+```
+
+Then choose the environnement of your choice. Likely to be `island-is-development01`. You be prompted for the following:
+
+```md
+CLI default client Region [eu-west-1]: <Press Enter>
+CLI default output format [json]: <Press Enter>
+CLI profile name [AWSPowerUserAccess-X]: <Custom name (e.g. dev)> or <Press Enter>
+```
+
+This step will add the new profile to your `~/.aws/config` file.
+
+2. Configure AWS, using the following configuration:
+
+```bash
+aws configure
+
+AWS Access Key ID [None]: <KEY_ID>
+AWS Secret Access Key [None]: <ACCESS_KEY>
+Default region name [eu-west-1]: <Press Enter>
+Default output format [json]: <Press Enter>
+```
+
+You will be able to find these information on the [AWS account](https://island-is.awsapps.com/start).
+
+Then, you will need to open the file created by AWS at `~/.aws/credentials` and add your session token to it.
+
+```bash
+[default]
+aws_access_key_id = <KEY_ID>
+aws_secret_access_key = <ACCESS_KEY>
+aws_session_token = <SESSION_TOKEN>
+```
```

Comment:
This is redundant just to fetch secrets. If you have AWS_PROFILE set to the profile you created in #1, the secret commands should work and use your SSO profile. The SSO profile is better since it works for 8 hours while this session token only works for 1 hour.

Reply:
If I'm only doing `yarn get-secrets api` for example, and even if I did `aws sso login --profile dev` beforehand, it will results to `An error occurred (ExpiredTokenException) when calling the GetParametersByPath operation: The security token included in the request is expired`. I need to explicitly pass the profile for the command to pick up my logged in so profile `AWS_PROFILE=dev yarn get-secrets api`

Reply:
Sorry, missed these replies.

Would this guide be simpler if we suggest saving the SSO profile as "default" (assuming that works), or suggest running `AWS_PROFILE=dev yarn get-secrets api`?